internal/pkg/git/git.go

```diff
@@ -12,40 +12,93 @@ import (
 	"github.com/keboola/keboola-as-code/internal/pkg/filesystem"
 	"github.com/keboola/keboola-as-code/internal/pkg/filesystem/aferofs"
 	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/template"
+	"github.com/keboola/keboola-as-code/internal/pkg/template/repository/manifest"
 )
 
 func Available() bool {
 	_, err := exec.LookPath("git")
 	return err == nil
 }
 
-func CheckoutTemplateRepository(url string, ref string, logger log.Logger) (filesystem.Fs, error) {
+func CheckoutTemplateRepository(ref model.TemplateRef, logger log.Logger) (filesystem.Fs, error) {
 	if !Available() {
 		return nil, fmt.Errorf("git command is not available, if you want to use templates from a git repository you have to install it first")
 	}
 
+	// Create a temp dir
 	dir, err := ioutil.TempDir("", "keboola-as-code-templates-")
 	if err != nil {
 		return nil, err
 	}
+	defer func(path string) {
+		_ = os.RemoveAll(path) // nolint: forbidigo
+	}(dir)
```

Comment:
Ten parameter `path`, je zbytocny, funkcia ma pristup do nadradeneho "namespace":
```
defer func() {
   _ = os.RemoveAll(dir) // nolint: forbidigo
}()
```

Reply:
Myslim, ze nie je dovod chybu zahodit ... moze sa zist pri debugovani:
```
        defer func() {
		if err = os.RemoveAll(dir); err != nil {
			logger.Warnf(`cannot remove temp dir "%s": %w`, dir, err)
		}
	}()
```

Reply:
Aha ok. To mi opravilo IDE, moc jsem nevěděl co s tím (ani kdy by ta chyba vlastně nastala a tím pádem co by se s ní mělo stát. Ale přišlo mi že to může být skoro jedno že nejde smazat, funkci to neovlivní. Warning zní dobře, upravím.